src/Util.ts

```diff
@@ -98,9 +99,58 @@ export default class W3Util {
     });
   }
 
-  public networkGasPrice(): Promise<BigNumber> {
+  public getNetworkId(): Promise<string> {
     return new Promise((resolve, reject) => {
-      this.web3.eth.getGasPrice((e: any, r: any) => (e ? reject(e) : resolve(r)));
+      this.web3.version.getNetwork((e: any, r: any) => (e ? reject(e) : resolve(r)));
+    });
+  }
+
+  public fetchExternalApiNetworkPrice (networkId: string): Promise<any> {
+    const services = SERVICES[networkId];
+    return new Promise(async(resolve, reject) => {
+      for ( let s = 0; s < services.length; s++) {
+        const service: IService = services[s]
+        try {
+          const data: any = await apiCall(service.api);
+          resolve(service.morph ? service.morph(data[service.field]) : data[service.field]);
+          break;
+        } catch (e) {
+          if (s === (services.length-1)) {
+            reject(new Error('No result from services'));
+          }
+          continue;
+        }
+      }
+    });
+  }
+
+  public externalApiNetworkPrice (): Promise<BigNumber> {
```

Comment:
Could make this an `async` function which would be the same thing as returning a `Promise`.

Reply:
I was just following the patterns and rules we have implemented, actually.